src/ucp/api/ucp.h

```diff
@@ -660,16 +664,30 @@ typedef struct ucp_ep_params {
     uint64_t                field_mask;
 
     /**
-     * Destination address; the address must be obtained using
-     * @ref ucp_worker_get_address.
-     * This field is mandatory for filling (along with its corresponding bit
+     * Destination address; if a connection manager isn't used,
+     * this address must be obtained using @ref ucp_worker_get_address and
+     * this field is mandatory for filling (along with its corresponding bit
      * in the field_mask - UCP_EP_PARAM_FIELD_REMOTE_ADDRESS).
-     * The ucp_ep_create routine will return with an error if the address isn't
+     * The ucp_ep_create routine will return with an error if the
+     * UCP_EP_PARAM_FIELD_ADDRESS_TYPE_CM flag isn't set and this address isn't
      * specified.
      */
     const ucp_address_t     *address;
 
     /**
+     * Destination address in the form of a sockaddr; if a connection manager
```

Comment:
i think it's better to add "flags" field to indicate connection establishment mode, and not use field_mask to influence the required behavior. each connection mode may require than some of the other fields will also be provided.